Document improvement Connect AWS (#1762)
thin-edge/thin-edge.io: docs/src/tutorials/connect-aws.md

certificate, then under the "Actions" dropdown click "Activate".

A policy needs to be attached to the device certificate in AWS IoT Core. AWS IoT Core policies determine what an
authenticated identity can do (here the authenticated identity is the device being connected). More info on AWS IoT Core
policies can be found [here](https://docs.aws.amazon.com/iot/latest/developerguide/iot-policies.html).

An example policy template for connecting a device running thin-edge can be found [here](./aws-example-policy.json).
To create a new policy, head over to the AWS IoT Core and navigate to "Security"->"Policies" then click "Create policy".
Give the policy a name, then under "Policy document" click "JSON", paste in the entire contents of the example policy
and click "Create". Be sure to replace `<region>`, `<account-id>`, etc. with correct values for your tenant.

The final step is to attach the policy to the certificate. Navigate to "Security"->"Certificates"
and click on the certificate that was uploaded previously. Click on "Attach policies" then select the policy that you

> created and click "Attach policies".

Reviewer: I would not remove all these notes as the instructions give no context.

Something along these lines:

```
The next step is to register the device on AWS IoT Hub, i.e, using the AWS terminology, . to create a Thing on AWS IoT hub. For a device to be trusted by AWS, one needs:
- to add the self-signed certificate to the AWS IoT Core (generated when using the `tedge cert create` command). 
- to attach a policy to the device certificate.

AWS IoT Core policies determine what an authenticated identity can do (here the authenticated identity is the device being connected). More info on AWS IoT Core policies can be found [here](https://docs.aws.amazon.com/iot/latest/developerguide/iot-policies.html).

An example policy template for connecting a device running thin-edge can be found [here](./aws-example-policy.json).
```

Author: My problem is that I cannot define from the existing one what is relevant and what not because it is not "usable", following these instructions will not lead you to connect thin-edge with AWS. I do agree that my proposal is not what we should have but with Reuben we have agreed that the intention is not to make an "Tutorial for AWS" because we don't want to maintain all our documentations if AWS, Azure etc changes something. Of course I can adjust the Instructions to desired ones, lets agree how that should look like

Reviewer 2: Since AWS can change their UI at anytime when they like, in my opinion, having a description only where to click is potentially fragile. I think we should focus on **WHAT** users need to create on AWS, not **HOW** to create.
However, if we describes steps to do something (HOW), bullet points are much more readable than writing in paragraph. I see Gligor's change has value on this point.

Also, we should have a blog post to describe what user needs to do on AWS side with nice screenshots ;)

My suggestion is how to write are below.
```
The first step is to create a policy.            # This is WHAT we need

[Steps (as of March 2023)]
1. Security -> Policies -> Create policy
2. Create policy -> Policy properties -> Policy name -> Enter the name of your policy (e.g.tedge)
3. Policy statements -> JSON -> Enter policy at (Example policy can be found [here](https://github.com/thin-edge/thin-edge.io/blob/2a4c963ccdab8d92a0102ebcd0d569df1ded8635/docs/src/tutorials/aws-example-policy.json)) Policy document -> Create

Next, create a Thing with the certificate that you generated with thin-edge.io in the previous section.

[Steps]
1. Manage -> All devices -> Things -> Create things
2. Create Single thing -> Next
3. Enter Thing name which can be obtain from the device with: tedge config get device.id
4. Device Shadow -> Unnamed shadow (classic) -> Next
5. Device certificate -> Use my certificate -> CA is not registered with AWS IoT
6. Certificate -> Choose file -> Select your tedge-certificate.pem file -> Open -> Next
7. Attach policies to certificate -optional -> Select your created policy -> Create thing

Now all preparations are done for the preparation on your AWS IoT side!
```

## Register the device on AWS IoT Hub

> For a device to be trusted by AWS, one needs to generate it by using the `tedge cert create` command.

Reviewer: To be precise one needs a device certificate not necessarily a self-signed.

 ```suggestion
For a device to be trusted by AWS, one needs a device certificate and the `tedge cert create` command is the simplest way to get one.
```

Also a policy needs to be attached to the device certificate in AWS IoT Core. AWS IoT Core policies determine what an
authenticated identity can do (here the authenticated identity is the device being connected). More info on AWS IoT Core

> policies can be found [here](https://docs.aws.amazon.com/iot/latest/developerguide/iot-policies.html).

Reviewer: I would add the link to the policy example here, adding these lines:

```
An example policy template for connecting a device running thin-edge can be found [here](./aws-example-policy.json). Be sure to replace `<region>`, `<account-id>`, etc. with correct values for your tenant.
```

To create a new policy, head over to the AWS IoT Core and navigate to ***Security -> Policies -> Create policy -> Policy properties -> Policy name -> Enter the name of your policy (e.g.tedge)***

> On the Policy statements tab Click on ***JSON*** and enter the policy in the ***Policy document*** (Example policy can be found [here](./aws-example-policy.json)) then click ***Create***

Reviewer 2: ```suggestion
On the Policy statements tab, click on ***JSON*** and enter the policy in the ***Policy document*** (an example policy can be found [here](./aws-example-policy.json)) then click ***Create***.
```

Reviewer: I would give the link to a policy example before the detailed steps.

> In the AWS IoT Core, navigate to ***Manage -> All devices -> Things -> Create things -> Create Single thing -> Next*** Enter Thing name which can be obtain from the device with: `tedge config get device.id`

Reviewer 2: ```suggestion
In the AWS IoT Core, navigate to ***Manage -> All devices -> Things -> Create things -> Create Single thing -> Next***. Enter Thing name which can be obtain from the device with: `tedge config get device.id`
```

> In the AWS IoT Core, navigate to ***Manage -> All devices -> Things -> Create things -> Create Single thing -> Next***. Enter Thing name which can be obtain from the device with: `tedge config get device.id`

Reviewer 3: ```suggestion
In the AWS IoT Core, navigate to ***Manage -> All devices -> Things -> Create things -> Create Single thing -> Next***. Enter the Thing name which can be obtained from the device with: `tedge config get device.id`
```

> In the Device Shadow section which allow connected devices to sync states with AWS choose ***Unnamed shadow (classic)*** and click ***Next*** and ***Configure device certificate - optional*** page opens

Reviewer 2: ```suggestion
In the Device Shadow section which allow connected devices to sync states with AWS, choose ***Unnamed shadow (classic)*** and click ***Next*** and ***Configure device certificate - optional*** page opens.
```

> In the Device Shadow section which allow connected devices to sync states with AWS choose ***Unnamed shadow (classic)*** and click ***Next*** and ***Configure device certificate - optional*** page opens.

Reviewer 3: ```suggestion
In the Device Shadow section which allows connected devices to sync states with AWS choose ***Unnamed shadow (classic)*** and click ***Next*** and ***Configure device certificate - optional*** page opens.
```

> At ***Device certificate*** choose ***Use my certificate -> CA is not registered with AWS IoT*** then ***Choose file*** snd select your tedge-certificate.pem file, click on ***Open -> Next***

Reviewer 2: ```suggestion
At ***Device certificate*** choose ***Use my certificate -> CA is not registered with AWS IoT*** then ***Choose file*** snd select your tedge-certificate.pem file, click on ***Open -> Next***.
```

> At ***Device certificate*** choose ***Use my certificate -> CA is not registered with AWS IoT*** then ***Choose file*** snd select your tedge-certificate.pem file, click on ***Open -> Next***.

Reviewer 3: ```suggestion
At ***Device certificate*** choose ***Use my certificate -> CA is not registered with AWS IoT*** then ***Choose file*** and select your tedge-certificate.pem file, click on ***Open -> Next***.
```

> Last step needed is to attach previously created policy to your certificate, ***Attach policies to certificate -optional*** -> Select your created policy ***-> Create thing***

Reviewer 2: ```suggestion
Last step needed is to attach previously created policy to your certificate, ***Attach policies to certificate -optional*** -> Select your created policy ***-> Create thing***.
```

> Last step needed is to attach previously created policy to your certificate, ***Attach policies to certificate -optional*** -> Select your created policy ***-> Create thing***.

Reviewer 3: ```suggestion
The last step needed is to attach the previously created policy to your certificate, ***Attach policies to the certificate -optional*** -> Select your created policy ***-> Create thing***.
```
